prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -111,10 +132,111 @@ class DeliveryFunction(
             name = "getDeliveryDetails",
             methods = [HttpMethod.GET],
             authLevel = AuthorizationLevel.ANONYMOUS,
-            route = "waters/report/{deliveryId}/delivery"
+            route = "waters/report/{id}/delivery"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("id") id: String,
+    ): HttpResponseMessage {
+        return this.getDetailedView(request, id)
+    }
+
+    /**
+     * Get a sortable list of delivery facilities
+     *
+     * @param request HTML request body.
+     * @param id Report or Delivery id for the report to get facilities from
+     * @return JSON of the facility list or errors.
+     */
+    @FunctionName("getDeliveryFacilities")
+    fun getDeliveryFacilities(
+        @HttpTrigger(
+            name = "getDeliveryFacilities",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "waters/report/{id}/facilities"
         ) request: HttpRequestMessage<String?>,
-        @BindingName("deliveryId") deliveryId: String,
+        @BindingName("id") id: String,
     ): HttpResponseMessage {
-        return this.getDetailedView(request, deliveryId)
+        try {
+            // Do authentication
+            val authResult = this.authSingleBlocks(request, id)
+            if (authResult != null)
+                return authResult
+
+            val params = HistoryApiParameters(request.queryParameters)
+            val facilityParams = FacilityListApiParameters(request.queryParameters)
+
+            val facilities = deliveryFacade.findDeliveryFacilities(
```

Comment:
Are we going to allow anyone to see everyone's deliveries?

Reply:
The call to `authSingleBlocks` earlier in the function does all the authorization logic. The auth part of the spec doesn't specify who can view which facilities, so I went with the same auth as the Detail view.